crates/router/src/connector/paybox/transformers.rs

```diff
@@ -102,121 +436,269 @@ impl From<PayboxPaymentStatus> for enums::AttemptStatus {
+    pub transaction_number: String,
+
+    #[serde(rename = "NUMAPPEL")]
+    pub call_number: String,
+
+    #[serde(rename = "CODEREPONSE")]
+    pub response_code: String,
+
+    #[serde(rename = "COMMENTAIRE")]
+    pub response_message: String,
+
+    #[serde(rename = "STATUS")]
+    pub status: PayboxStatus,
 }
 
-impl<F, T>
-    TryFrom<types::ResponseRouterData<F, PayboxPaymentsResponse, T, types::PaymentsResponseData>>
+impl<F, T> TryFrom<types::ResponseRouterData<F, PayboxResponse, T, types::PaymentsResponseData>>
     for types::RouterData<F, T, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: types::ResponseRouterData<F, PayboxPaymentsResponse, T, types::PaymentsResponseData>,
+        item: types::ResponseRouterData<F, PayboxResponse, T, types::PaymentsResponseData>,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            status: enums::AttemptStatus::from(item.response.status),
-            response: Ok(types::PaymentsResponseData::TransactionResponse {
-                resource_id: types::ResponseId::ConnectorTransactionId(item.response.id),
-                redirection_data: None,
-                mandate_reference: None,
-                connector_metadata: None,
-                network_txn_id: None,
-                connector_response_reference_id: None,
-                incremental_authorization_allowed: None,
-                charge_id: None,
+        let response = item.response.clone();
+        let status = get_status_of_request(response.response_code.clone());
+        match status {
+            true => Ok(Self {
+                response: Ok(types::PaymentsResponseData::TransactionResponse {
+                    resource_id: types::ResponseId::ConnectorTransactionId(response.call_number),
+                    redirection_data: None,
+                    mandate_reference: None,
+                    connector_metadata: Some(serde_json::json!(PayboxMeta {
+                        connector_request_id: response.transaction_number.clone()
+                    })),
+                    network_txn_id: None,
+                    connector_response_reference_id: Some(response.transaction_number),
+                    incremental_authorization_allowed: None,
+                    charge_id: None,
+                }),
+                ..item.data
             }),
-            ..item.data
-        })
+            false => Ok(Self {
+                response: Err(types::ErrorResponse {
+                    code: response.response_code.clone(),
+                    message: response.response_message.clone(),
+                    reason: Some(response.response_message),
+                    status_code: item.http_code,
+                    attempt_status: None,
+                    connector_transaction_id: Some(item.response.transaction_number),
+                }),
+                ..item.data
+            }),
+        }
```

Comment:
This implementation wont work for Capture flow.
Separate this handle response implementation for authorize and capture flow.